editor.planx.uk/src/pages/FlowEditor/components/Settings/Submissions/EventsLog.tsx

```diff
@@ -93,6 +97,20 @@ const EventsLog: React.FC<GetSubmissionsResponse> = ({
 const CollapsibleRow: React.FC<Submission> = (submission) => {
   const [open, setOpen] = useState<boolean>(false);
 
+  const [teamSlug, canUserEditTeam, submissionEmail] = useStore((state) => [
+    state.teamSlug,
+    state.canUserEditTeam,
+    state.teamSettings?.submissionEmail,
+  ]);
+
+  // Only show an application download button if certain conditions are met
+  const submissionDataExpirationDate = addDays(new Date(), DAYS_UNTIL_EXPIRY);
+  const showDownloadButton =
+    canUserEditTeam(teamSlug) &&
+    submission.status === "Success" &&
+    submissionEmail &&
```

Comment:
It's admittedly a bit of a weird requirement that `submissionEmail` needs to be set to be able to download application data that was originally sent to any back-office system (because `submissionEmail` is "send to email" specific) - but now that this is self-service in the editor, I think it's ultimately okay and won't be a blocker?

This MVP re-uses the `/download-application-files/:sessionId` endpoint, but as other comments here also say we know we want to change/"harden" this soon so will be able to theoretically drop this requirement in next iteration!

Reply:
Yeah it's a bit odd as it stands but I totally get it!

I think a better approach would be to hit the API and use the user's JWT permissions to control access as opposed to the fact they have access to the `submission_email` column. 

The reason for this is that `team_settings` doesn't have access controls between teams, but the `lowcal_sessions` table could / should.

What I think this would look like - 
 - Set up permissions on `lowcal_sessions` table so that `teamEditors` can access sessions from their team
 - Make a new route, with middleware to check it's an authenticated request, but use the same controller (might need to look at which client is making the request?)
 - Hit the new route here

```ts
router.get("/download-submission-files/:sessionId", useTeamEditorAuth, downloadApplicationFiles);
```

This works fine for now, but when we revisit this I think it's worth tightening this up a little 👍